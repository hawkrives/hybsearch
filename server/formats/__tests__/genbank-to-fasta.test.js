/* eslint-env jest */

const genbankToFasta = require('../genbank-to-fasta')
const hashNames = require('../fasta/hash-names')
const fs = require('fs')
const path = require('path')

const base = path.join(__dirname, 'input')
const files = fs.readdirSync(base).filter(f => f.endsWith('.gb'))

for (const file of files) {
  test(file, () => {
    const content = fs.readFileSync(path.join(base, file), 'utf-8')
    const tree = genbankToFasta(content, hashNames(content))
    expect(tree).toMatchSnapshot()
  })
}
