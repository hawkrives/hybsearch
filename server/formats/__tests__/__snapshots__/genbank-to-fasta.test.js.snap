// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`accounts for multiple spaces in the species name 1`] = `
"> Panthera_pardus__AB817079
ccatccaacatctcaacatgatggaactttggctccctattaggggtatgtttaatccta"
`;

exports[`felidae.gb 1`] = `
"> Panthera_pardus__AB817079
ccatccaacatctcaacatgatggaactttggctccctattaggggtatgtttaatccta
caaattctcaccggcctctttctagccatacattatacatcagacacaacaaccgctttc
tcatcagttacccatatctgccgcgatgtaaattatggctgaattatccggtatctacac
gccaatggagcctccatattctttatctgcctatacatacatgtaggacgagggatatac
tatggctcctacactttctcagagacatggaacattggagtcgtattattgttcacggtt
atggccacagccttcataggatatgtcttaccgtggggccaaatatccttttgaggggca
accgtaattaccaacctcctatcagcaatcccatacattgggattgacctagtagaatga
atctgagggggcttctcagtagacaaagctaccttgacacgattctttgccttccacttc
atccttccatttatcatctcagctctagcagcagtccacctcctattccttcacgagaca
ggatctaacaacccctcaggaatagtatctgactcagacaaaattccattccacccatac
tacacaatcaaagatatcctgggccttctagtactaatcttagcactcatactactcgtc
ctattctcaccagacctgttaggagaccccgataactacatccccgccaaccctctaaat
acccctccccatatcaagcctgaatgatacttcctatttgcatacgcaatcctccgatcc
attcccaacaaactaggaggagtcctagccctagtcctatccatcctaatcttagcagtc
atcccc
> Panthera_pardus__AB817078
ccatccaacatctcaacatgatggaactttggctccctattaggggtatgtttaatccta
caaattctcaccggcctctttctagccatacattatacatcagacacaacaaccgctttc
tcatcagttacccatatctgccgcgatgtaaattatggctgaattatccggtatctacac
gccaatggagcctccatattctttatctgcctatacatacatgtaggacgagggatatac
tatggctcctacactttctcagagacatggaacattggagtcgtattattgttcacggtt
atggccacagccttcataggatatgtcttaccgtggggccaaatatccttttgaggggca
accgtaattaccaacctcctatcagcaatcccatacattgggattgacctagtagaatga
atctgagggggcttctcagtagacaaagctaccttgacacgattctttgccttccacttc
atccttccaattatcatctcagctctagcagcagtccacctcctattccttcacgagaca
ggatctaacaacccctcaggaatagtatctgactcagacaaaattccattccacccatac
tacacaatcaaagatatcctgggccttctagtactaatcttagcactcatactactcgtc
ctattctcaccagacctgttaggagaccccgataactacatccccgccaaccctctaaat
acccctccccatatcaagcctgaatgatacttcctatttgcatacgcaatcctccgatcc
attcccaacaaactaggaggagtcctagccctagtcctatccatcctaatcttagcagtc
atcccc
"
`;

exports[`handles URLs with // in them 1`] = `
"> Panthera_pardus__AB817079
ccatccaacatctcaacatgatggaactttggctccctattaggggtatgtttaatccta"
`;

exports[`vulpes.gb 1`] = `
"> Vulpes_macrotis__AH005827
ctcacccactagctaaaatcgttaacgactcattcattgacctccccgcaccatcaaaca
tttctgcctgatgaaacttcggatccctcctaggtatctgcctcattctacagattataa
caggcctatttctagccatacactacacatccgatacagctactgctttttcatcagtta
cacacatttgccgagacgttaactacggctgaatcattcgctacatacatgcaaatggag
catctatattcttcatttgtctgttcatgcacgtgggacgaggcctatattatggatcct
acgtattcatagaaacatgaaacgtcggaattgtcttactattcgcaaccatagccannn
nnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnn
nnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnncccctccacatatcaaaccctga
gtgatacttcctattcgcctacgctattctccggtccatccctaacaaactagggggagt
cctcgccctagtattctccatcctaatcttagcactcatccctcacctccatacctcaaa
acaacgcggaataatattccgacccctcagtcaatgcctattctgactcctaaccgcaga
cctcctcaccttaacctgaatcggaggacagccagtagaacacccctttatcatcatcgg
ccaaatcgcctctatcctatactttgctatcttattgattctaataccaaccattagcat
tattgaaaataatctcctaaagtgaag
"
`;
